### ISSUE TITLE
How to deal with the overfitting and underfitting problems during the training loop?

### ISSUE THREAD

**Issue by Wenhao-Yang** [Wenhao-Yang/PythonLearning#1]
As far as I am concerned, there are several aspects that should be mentioned.

1. The number of variables in the model.

2. The learning rate during the training.

3. The  number of iterations.

....
to be continued
